kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/StorageWriteApiValidator.java

```diff
@@ -0,0 +1,65 @@
+package com.wepay.kafka.connect.bigquery.config;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.DELETE_ENABLED_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.ENABLE_BATCH_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.ENABLE_BATCH_MODE_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.UPSERT_ENABLED_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.USE_STORAGE_WRITE_API_CONFIG;
+
+public class StorageWriteApiValidator extends MultiPropertyValidator<BigQuerySinkConfig> {
+
+    private static final Collection<String> DEPENDENTS = Collections.unmodifiableCollection(Arrays.asList(
+            UPSERT_ENABLED_CONFIG, DELETE_ENABLED_CONFIG, ENABLE_BATCH_CONFIG
+    ));
+
+    protected StorageWriteApiValidator(String propertyName) {
+        super(propertyName);
+    }
+
+    protected StorageWriteApiValidator() {
+        super(USE_STORAGE_WRITE_API_CONFIG);
+    }
+
+    @Override
+    protected Collection<String> dependents() {
+        return DEPENDENTS;
+    }
+
+    @Override
+    protected Optional<String> doValidate(BigQuerySinkConfig config) {
+        if (!config.getBoolean(USE_STORAGE_WRITE_API_CONFIG)) {
+            if (config.getBoolean(ENABLE_BATCH_MODE_CONFIG)) {
+                return Optional.of("Storage Write Api Batch load is supported only when "
+                        + USE_STORAGE_WRITE_API_CONFIG + " is enabled. Either disable batch mode or enable Storage Write" +
+                        " API");
```

Comment:
Minor suggestion - Lets move the config validation error strings to some constant string variables and use them here. 

Also we can use the same strings in UT as well.